src/agent/config.ts

```diff
@@ -257,6 +257,11 @@ export interface ResolvedDebugAgentConfig extends GoogleAuthOptions {
    * Uses by tests to cause the start() function to return the debuglet.
    */
   testMode_: boolean;
+
+  /**
+   * used by tests that nock to isolate nocks to their own scopes.
+   */
+  apiUrl?: string;
```

Comment:
Since this is only used by tests we should name it `apiUrl_` to indicate that its private.

Reply:
this is a really great nit. ill fix it up

this change was largely aspirational because right now the api url is defined as a constant in the controller class. I really think it would be better if all of the defaults come from config with a convenient api for setting values.